app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -111,11 +111,13 @@ def identity_role
       # If there is no jwt identity so role and username are nil
       def audit_role_id
         return @audit_role_id if @audit_role_id
+        # Dont change to jwt_identity so function won't be called if validate_and_decode failed. Because we
+        # won't have jwt token where we can fetch the identity from
```

Comment:
that doesnt sound right, if this is the logic, than `jwt_identity` is supposed to implement it. if it stays that way it can easily become a bug

Reply:
jwt_identity need jwt decoded token to work. If validate and decode fails there is no decoded token so there is no identity and jwt_identity can't run.

authenticator.rb know if validate_and_decode runed or not. jwt_identity don't knows and dont need to know. There fore if valdiate and decode didn't run so in the audit failiure we should throw false.